### PR TITLE
perf: avoid in place operator where ineffective

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This commit enhances performance by utilizing the `c = a op b` style instead of `c = a; c inplace-op b`. While the former style is nearly as performant as the latter when `a` is movable, it becomes inefficient when operating on prime fields or points on elliptic curves that are not able to be moved. This change ensures better performance and readability, reducing 2 lines to 1. Note that irrelevant changes may occur due to this adjustment.

### CODE REVIEW

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Review by Insun35 — APPROVED** [review]
LGTM

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

File: tachyon/math/elliptic_curves/short_weierstrass/projective_point_impl.h [hunk 1]
```diff
@@ -155,60 +135,48 @@ constexpr CLASS& CLASS::DoubleInPlace() {
...
-  BaseField zz = z_;
-  zz.SquareInPlace();
+  BaseField zz = z_.Square();
 
   // w = a * ZZ + 3 * XX
-  BaseField w = xx;
-  w += w.Double();
+  BaseField w = xx + xx.Double();
   if constexpr (!Curve::Config::kAIsZero) {
     w += Curve::Config::MulByA(zz);
```
Comment (ashjeong) [inline tachyon/math/elliptic_curves/short_weierstrass/projective_point_impl.h:146]:
```suggestion
  BaseField w = xx.Double();
  w += xx;
  if constexpr (!Curve::Config::kAIsZero) {
    w += Curve::Config::MulByA(std::move(zz));

File: tachyon/math/finite_fields/fp12.h [hunk 2]
```diff
@@ -282,8 +279,7 @@ class Fp12 final : public QuadraticExtensionField<Fp12<Config>> {
     this->c1_ -= b;
     // c0 = ((α₃β₃ + α₅β₄q) + (α₄β₃ + α₃β₄)x + (α₅β₃ + α₄β₄)x²)p +
     //      α₀β₀ + α₁β₀x + α₂β₀x², where p = y² and q = x³
-    this->c0_ = Config::MulByNonResidue(b);
-    this->c0_ += a;
+    this->c0_ = Config::MulByNonResidue(b) + a;
```
Comment (ashjeong) [inline tachyon/math/finite_fields/fp12.h:282]: I don't think this should be changed:
its equal to the case 
```c++
z = b * c + a
```
which should be treated as such:
```c++
z = b * c;
z += a;
```

 REF[0] for future similar scenarios

File: tachyon/math/finite_fields/fp12.h [hunk 3]
```diff
@@ -342,8 +335,7 @@ class Fp12 final : public QuadraticExtensionField<Fp12<Config>> {
     // c0 = (α₅β₄q + α₃β₄x + α₄β₄x²)p +
     //      (α₀β₀ + α₂β₁q) + (α₁β₀ + α₀β₁)x + (α₂β₀ + α₁β₁)x²,
     //      where p = y² and q = x³
-    this->c0_ = Config::MulByNonResidue(b);
-    this->c0_ += a;
+    this->c0_ = Config::MulByNonResidue(b) + a;
```
Comment (ashjeong) [inline tachyon/math/finite_fields/fp12.h:338]: refer to REF[0]

**Review by ashjeong — CHANGES_REQUESTED** [review]
I tried to only include small fixes/ changes related to the commit, yet there appears to be a lot of redundant or unneeded variables or calculations. Perhaps a separate pr may be needed to fix these.

File: tachyon/math/finite_fields/fp6.h [hunk 4]
```diff
@@ -315,29 +315,15 @@ class Fp6<Config, std::enable_if_t<Config::kDegreeOverBaseField == 3>> final
...
-    this->c0_ = std::move(t0);
-    // c1 = α₀β₁
-    this->c1_ = std::move(t1);
     // c2 = α₁β₁
-    this->c2_ = std::move(t2);
+    this->c2_ = this->c1_ * beta1;
+    // c1 = α₀β₁
+    this->c1_ = this->c0_ * beta1;
+    // c0 = α₂β₁q
+    this->c0_ = Config::MulByNonResidue(t0);
```
[the remark this answers is not in the document]
Comment (chokobole) [reply tachyon/math/finite_fields/fp6.h:326]: We can't do that, `c0` is used to compute `c1`, and `c1` is used to compute `c2` so we need to do in this order.

File: tachyon/math/elliptic_curves/short_weierstrass/projective_point_impl.h (hunk 1 above)
Comment (chokobole) [reply tachyon/math/elliptic_curves/short_weierstrass/projective_point_impl.h:146]: Actually `MulByA()` takes const reference as an argument!

File: tachyon/math/finite_fields/fp6.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (ashjeong) [reply tachyon/math/finite_fields/fp6.h:326]: oops yea didn't catch that

**Pull request merged**